projects/user_publication/publications.py

```diff
@@ -93,48 +94,150 @@ def choose_approved_with_option():
             "if email is already sent then choose this option to input the confirmation"
         )
     )
-    print("4. To leave the status as is. Choose this")
-    inp = input("Choose 1, 2, 3 or 4")
+    print("4. If the publication did not use chameleon")
+    print("5. To leave the status as is. Choose this")
+    inp = input("Choose 1, 2, 3 or 4: ")
     if inp == "1":
         return PublicationSource.APPROVED_WITH_PUBLICATION
     elif inp == "2":
-        print("You chose two")
         return PublicationSource.APPROVED_WITH_JUSTIFICATION
     elif inp == "3":
         return PublicationSource.APPROVED_WITH_EMAIL
     elif inp == "4":
-        return
+        return 
+    elif inp == "5":
+        return PublicationSource.APPROVED_WITH_PENDING_REVIEW
     else:
         return choose_approved_with_option()
 
 
+def choose_email_approval_status():
+    print("Change the status of approving the publication with email")
+    print("1. If attestation is received")
+    print("2. If rejected")
+    print("3. To leave as is, no reply yet from the user")
+    inp = input("Choose 1, 2, 3: ")
+    if inp == "1":
+        return 'approved'
+    elif inp == "2":
+        return 'rejected'
+    elif inp == "3":
+        return
+    else:
+        return choose_email_approval_status()
+
+
+def update_status_for_email_approval(pub, user_reported_source):
+    email_choice = choose_email_approval_status()
+    if email_choice == 'approved':
+        with transaction.atomic():
+            user_reported_source.approved_with = PublicationSource.APPROVED_WITH_EMAIL
+            user_reported_source.save()
+            pub.status = Publication.STATUS_APPROVED
+            pub.save()
+    elif email_choice == 'rejected':
+        with transaction.atomic():
+            user_reported_source.approved_with = PublicationSource.APPROVED_WITH_EMAIL

```

Comment:
This should not be set if the publication is rejected.

Reply:
yeah. It is confusing. The meaning of this is, it was tried for approval with email and then got rejected from email.
This helps us maintain a count of how many got rejected from email.

Reply:
I don't think we should be using the `approved_with` field to describe how a publication is rejected. I think this should be left null. 

If a publication is rejected, it's basically for two reasons:

1. It's very obvious (published in 1999 or something like that)
2. I emailed the PI and they said they didn't use Chameleon